dev/com.ibm.ws.microprofile.metrics.2.3.monitor/src/com/ibm/ws/microprofile/metrics/monitor/MonitorMetrics.java

```diff
@@ -29,187 +29,213 @@
+				Tr.debug(tc, "Registered " + metricID.toString());
+			} else if (MetricType.GAUGE.equals(type)) {
+				MonitorGauge<Number> mg = metricData[MappingTable.MBEAN_SUBATTRIBUTE] == null
+						? new MonitorGauge<Number>(mbs, objectName, metricData[MappingTable.MBEAN_ATTRIBUTE])
+						: new MonitorGauge<Number>(mbs, objectName, metricData[MappingTable.MBEAN_ATTRIBUTE],
+								metricData[MappingTable.MBEAN_SUBATTRIBUTE]);
+				metricRegistry.register(Metadata.builder().withName(metricName)
+						.withDisplayName(metricData[MappingTable.METRIC_DISPLAYNAME])
+						.withDescription(metricData[MappingTable.METRIC_DESCRIPTION]).withType(type)
+						.withUnit(metricData[MappingTable.METRIC_UNIT]).build(), mg, metricTag);
+				metricIDSet.add(metricID);
+				Tr.debug(tc, "Registered " + metricID.toString());
+			}
+			// Only REST_Stats is using SIMPLETIMER at the moment
+			else if (MetricType.SIMPLE_TIMER.equals(type)) {
+
+				MonitorSimpleTimer mst = new MonitorSimpleTimer(mbs, objectName,
+						metricData[MappingTable.MBEAN_ATTRIBUTE], metricData[MappingTable.MBEAN_SUBATTRIBUTE],
+						metricData[MappingTable.MBEAN_SECOND_ATTRIBUTE],
+						metricData[MappingTable.MBEAN_SECOND_SUBATTRIBUTE]);
+				String[] objName_rest = getRESTMBeanStatsTags();
+
+				// Only REST STAT metric is using SIMPLE TIMER.. explicitly creating the tags
+				// necessary.
+				String appName = objName_rest[0];
+				Tag classTag = new Tag("class", objName_rest[1]);
+				Tag methodTag = new Tag("method", objName_rest[2]);
+
+				metricRegistry.register(Metadata.builder().withName(metricName)
+						.withDisplayName(metricData[MappingTable.METRIC_DISPLAYNAME])
+						.withDescription(metricData[MappingTable.METRIC_DESCRIPTION]).withType(type)
+						.withUnit(metricData[MappingTable.METRIC_UNIT]).build(), mst, classTag, methodTag);
+
+				metricID = new MetricID(metricName, classTag, methodTag);
+				metricIDSet.add(metricID);
+				sharedMetricRegistry.associateMetricIDToApplication(metricID, appName, metricRegistry);
+
+				Tr.debug(tc, "Registered " + metricID.toString());
+			} else {
+				Tr.debug(tc, "Failed to register " + metricName + " because of invalid type " + type);
+			}
+
+			// reset
+			metricRegistry = null;
+		}
 	}
 
 	protected String[] getRESTMBeanStatsTags() {
-		String[] Tags = new String[3];
+		String[] mbeanProperties = new String[3];
 		for (String subString : objectName.split(",")) {
 			subString = subString.trim();
-            if (subString.contains("name=")) {
-            	Tags = subString.split("/");
-            	//blank method
-            	Tags[2] = Tags[2].replaceAll("\\(\\)", "");
-            	//otherwise first bracket becomes underscores
-            	Tags[2] = Tags[2].replaceAll("\\(", "_");
-            	//second bracket is removed
-            	Tags[2] = Tags[2].replaceAll("\\)", "");
-            	break;
-            }
+
+			if (subString.contains("name=")) {
+				mbeanProperties = subString.split("/");
```

Comment:
Not sure the format is.  Is it safe to assume you will always have 3 elements?

Reply:
Yes, it's expected to be:
`name=ApplicationName/fully.qualified.class.name/methodSignature(java.lang.String)`

This is the Mbean property constructed from the JAXRS filter.

Have made a commit with a comment, and renamed the the variable.